public/javascripts/game.js

```diff
@@ -142,177 +170,245 @@
-//				console.log("");
-//
-//				console.log("VELOCITY VECTOR: ");
-//				console.log("Vx: " + circle.velX);
-//				console.log("Vy: " + circle.velY);
-//				console.log("");
-
-				var tx =1.0;
-				var ty=0;
-				var dot = (rx*tx) + (ry*ty);
-				updateInput(inputs,0, dot,distance);
-
-
-
-				var rad = (45*pi())/180;
-				var cs = Math.cos(rad);
-				var sn = Math.sin(rad);
-
-				var px = tx * cs - ty * sn;
-				var py = tx * sn + ty * cs;
-
-
-				dot = (rx*px) + (ry*py);
-				updateInput(inputs,1, dot,distance);
-
-
-				rad = (90*pi())/180;
-				cs = Math.cos(rad);
-				sn = Math.sin(rad);
-
-				px = tx * cs - ty * sn;
-				py = tx * sn + ty * cs;
-
-
-				dot = (rx*px) + (ry*py);
-				updateInput(inputs,2, dot,distance);
-				
-				
-				
-				rad = (135*pi())/180;
-				cs = Math.cos(rad);
-				sn = Math.sin(rad);
-				px = tx * cs - ty * sn;
-				py = tx * sn + ty * cs;
-				dot = (rx*px) + (ry*py);
-				updateInput(inputs,3, dot,distance);
 				
 				
-				rad = (180*pi())/180;
-				cs = Math.cos(rad);
-				sn = Math.sin(rad);
-				px = tx * cs - ty * sn;
-				py = tx * sn + ty * cs;
-				dot = (rx*px) + (ry*py);
-				updateInput(inputs,4, dot,distance);
-				
-				rad = (225*pi())/180;
-				cs = Math.cos(rad);
-				sn = Math.sin(rad);
-				px = tx * cs - ty * sn;
-				py = tx * sn + ty * cs;
-				dot = (rx*px) + (ry*py);
-				updateInput(inputs,5, dot,distance);
-				
-				
-				rad = (270*pi())/180;
-				cs = Math.cos(rad);
-				sn = Math.sin(rad);
-				px = tx * cs - ty * sn;
-				py = tx * sn + ty * cs;
-				dot = (rx*px) + (ry*py);
-				updateInput(inputs,6, dot,distance);
-				
-				
-				rad = (315*pi())/180;
-				cs = Math.cos(rad);
-				sn = Math.sin(rad);
-				px = tx * cs - ty * sn;
-				py = tx * sn + ty * cs;
-				dot = (rx*px) + (ry*py);
-				updateInput(inputs,7, dot,distance);
-				
-				
-
-			}
-			if(flag==1){
-				continue;
+				if(distance<minDist){
+					minDist=distance;
+					foodx=xDist/distance;
+					foody=yDist/distance;
+				}
+		
 			}
 			
-			//console.log(inputs);
+			
+			
+//			var netInputs = [(inputs[0]/((w)*Math.sqrt(2))),
+//							 (inputs[1]/((w)*Math.sqrt(2))),
+//							 (inputs[2]/((w)*Math.sqrt(2))),
+//							 (inputs[3]/((w)*Math.sqrt(2))),
+//							 (inputs[4]/((w)*Math.sqrt(2))),
+//							 (inputs[5]/((w)*Math.sqrt(2))),
+//							 (inputs[6]/((w)*Math.sqrt(2))),
+//							 (inputs[7]/((w)*Math.sqrt(2)))];
 
-			//console.log(inputs);
 
-			//normalize inputs what range to we actually care about
-			//how far can the fish see infinite ???
 
-			var netInputs = [inputs[0]/((w/2)*Math.sqrt(2)),
-							 inputs[1]/((w/2)*Math.sqrt(2)),
-							 inputs[2]/((w/2)*Math.sqrt(2)),
-							 inputs[3]/((w/2)*Math.sqrt(2)),
-							 inputs[4]/((w/2)*Math.sqrt(2)),
-							 inputs[5]/((w/2)*Math.sqrt(2)),
-							 inputs[6]/((w/2)*Math.sqrt(2)),
-							 inputs[7]/((w/2)*Math.sqrt(2))];
+			var netInputs = [foodx,foody,circle.velX/fish_speed,circle.velY/fish_speed];
```

Comment:
A comment here about what these inputs to the net are could help readers avoid some unnecessary searching